### PR TITLE
FLW-1270 iOS: Migrate from Subscan to Coingecko API for USD representation of KSM & DOT

### PR DESCRIPTION
**Pull request by stas-litvinov**
Replace Subscan API with Coingecko API

### CODE REVIEW

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            )
+
+            request.httpMethod = HttpMethod.get.rawValue
+
+            return request
+        }
+
+        let resultFactory = AnyNetworkResultFactory<PriceData?> { data in
+            let priceData = try JSONDecoder().decode(
+                CoingeckoPriceData.self,
```
Comment (ERussel) [inline fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:92]: Why do we need to deal with dynamic keys if we could just parse as `[String: PriceData].self`?

Comment (stas-litvinov) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:92]: I tried, but it did not work. Later I found another problem with mapping and decided to leave it as it is now. I guess, we'll need it later

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    static let baseUrlString = "https://api.coingecko.com/api/v3"
+
+    private func buildURLForAssets(
+        _ assets: [WalletAssetId],
+        method: String,
+        currencies: [String] = ["usd"],
+        options: CoingeckoPriceRequestOptions = [.includeDayChange, .includeLastUpdatedAt]
+    ) -> URL? {
+        guard var components = URLComponents(
+            string: Self.baseUrlString + method
```
Comment (ERussel) [inline fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:36]: It is more safe to use `appendingPathComponent`

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import RobinHood
+
+protocol CoingeckoOperationFactoryProtocol {
+    func fetchPriceOperation(for assets: [WalletAssetId]) -> BaseOperation<[PriceData]>
+}
+
+struct CoingeckoPriceRequestOptions: OptionSet {
+    let rawValue: Int
+
+    static let includeMarketCap = CoingeckoPriceRequestOptions(rawValue: 1 << 0)
```
Comment (antonkhvorov) [inline fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:11]: We don't need it

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    static let includeDayVolume = CoingeckoPriceRequestOptions(rawValue: 1 << 1)
```
Comment (antonkhvorov) [inline fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:12]: We don't need it

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    static let includeDayChange = CoingeckoPriceRequestOptions(rawValue: 1 << 2)
+    static let includeLastUpdatedAt = CoingeckoPriceRequestOptions(rawValue: 1 << 3)
```
Comment (antonkhvorov) [inline fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift:14]: We don't need it

**Review by antonkhvorov — CHANGES_REQUESTED** [review]

**Comment by antonkhvorov** [comment]
Let keep things nice & simple, especially to the parts which are going to be reworked in 2.0 (such as this feature).
This PR should be straightforward & small, so that there will be no room for errors since we need to release it asap

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift (hunk 1 above)
Comment (ERussel) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift]: fixed

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift (hunk 2 above)
Comment (ERussel) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift]: fixed

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift (hunk 3 above)
Comment (ERussel) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift]: fixed

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift (hunk 4 above)
Comment (ERussel) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift]: fixed

File: fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift (hunk 5 above)
Comment (ERussel) [reply fearless/Common/Network/Coingecko/CoingeckoOperationFactory.swift]: fixed

**Review by antonkhvorov — APPROVED** [review]

**Pull request merged**